api/representation_test.go

```diff
@@ -0,0 +1,431 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul-terraform-sync/api/oapigen"
+	"github.com/hashicorp/consul-terraform-sync/config"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	taskReq = `{
+    "description": "Writes the service name, id, and IP address to a file",
+    "enabled": true,
+    "name": "new-example-task",
+    "providers": [
+        "local"
+    ],
+    "services": [
+        "api"
+    ],
+    "source": "./example-module",
+    "variable_files": [],
+    "buffer_period": {
+        "enabled": true,
+        "max": "0s",
+        "min": "0s"
+    }
+}`
+)
+
+func TestTaskRequest_String(t *testing.T) {
+	var req taskRequest
+
+	err := json.Unmarshal([]byte(taskReq), &req)
+	require.NoError(t, err)
+
+	actual := fmt.Sprintf("%s", req)
+	expected := "{\"buffer_period\":{\"enabled\":true,\"max\":\"0s\",\"min\":\"0s\"},\"description\":\"Writes the service name, id, and IP address to a file\",\"enabled\":true,\"name\":\"new-example-task\",\"providers\":[\"local\"],\"services\":[\"api\"],\"source\":\"./example-module\"}\n"
```

Comment:
if possible, it's easier to read if it's wrapped around string literal ` instead of double quotes which require escaping each quote, like the above taskReq

Reply:
Some background, the reason I included it like this, is because this is how `require.Equal(t, expected, actual)` returns the expected matching string when you don't match, so it was easy to copy and paste the expected match after verifying it visually.

However, that's not a good enough reason not to change it, so I've made your suggested change!

Reply:
I see! didn't realize the perk here, given that this is a nit and would be happy either way